dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/intermediate/components/IntegerComponent.kt

```diff
@@ -0,0 +1,68 @@
+package org.dataland.frameworktoolbox.intermediate.components
+
+import org.apache.commons.text.StringEscapeUtils
+import org.dataland.frameworktoolbox.intermediate.FieldNodeParent
+import org.dataland.frameworktoolbox.specific.datamodel.TypeReference
+import org.dataland.frameworktoolbox.specific.datamodel.elements.DataClassBuilder
+import org.dataland.frameworktoolbox.specific.fixturegenerator.elements.FixtureSectionBuilder
+import org.dataland.frameworktoolbox.specific.uploadconfig.elements.UploadCategoryBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+
+/**
+ * A IntegerComponent represents a numeric value from the integer domain
+ */
+open class IntegerComponent(
+    identifier: String,
+    parent: FieldNodeParent,
+) : ComponentBase(identifier, parent) {
+    open var constantUnitSuffix: String? = null
+
+    override fun generateDefaultDataModel(dataClassBuilder: DataClassBuilder) {
+        dataClassBuilder.addProperty(
+            this.identifier,
+            documentSupport.getJvmTypeReference(
+                TypeReference("java.math.BigInteger", isNullable),
+                isNullable,
+            ),
+        )
+    }
+
+    override fun generateDefaultViewConfig(sectionConfigBuilder: SectionConfigBuilder) {
+        sectionConfigBuilder.addStandardCellWithValueGetterFactory(
+            this,
+            documentSupport.getFrameworkDisplayValueLambda(
+                FrameworkDisplayValueLambda(
+                    "formatNumberForDatatable(${getTypescriptFieldAccessor(true)}," +
+                        " \"${StringEscapeUtils.escapeEcmaScript(constantUnitSuffix ?: "")}\")",
+                    setOf(
+                        "import { formatNumberForDatatable } from " +
+                            "\"@/components/resources/dataTable/conversion/NumberValueGetterFactory\";",
+                    ),
+                ),
+                label, getTypescriptFieldAccessor(),
+            ),
+        )
+    }
+
+    override fun generateDefaultUploadConfig(uploadCategoryBuilder: UploadCategoryBuilder) {
+        uploadCategoryBuilder.addStandardUploadConfigCell(
+            component = this,
+            uploadComponentName = "NumberFormField",
+            validation = "integer",
+            unit = constantUnitSuffix,
```

Comment:
Why is this unit field not also set in the DecimalComponent?

Reply:
I think noone noticed that it is missing in the DecimalComponent because there is no decimal field in this framework which has a unit.
But it makes sense to add the possibility to add a unit there too, right?

Shall I add it there too? What do you think? Personally I think we should do it.

Reply:
I think we are less prone ot bugs if unit is also set there too. Maybe a shared base class can handle this.

Reply:
DecimalComponent now also puts its unit into the uploadConfig entry.